jobsdb/jobsdb.go

```diff
@@ -1058,7 +1061,7 @@ Function to return an ordered list of datasets and datasetRanges
 Most callers use the in-memory list of dataset and datasetRanges
 Caller must have the dsListLock readlocked
 */
-func (jd *HandleT) getDSList() []dataSetT {
+func (jd *HandleT) GetDSList() []dataSetT {
```

Comment:
Not that comfortable with exporting this method, let alone when it has an unexported return type.

What about creating a `HandleInspector` in `jobsdb` package, e.g.

```
type HandleInspector struct {
	*HandleT
}

func (h *HandleInspector) DSListSize() int {
	return len(h.HandleT.getDSList())
}

```

Then you could use this inspector from your tests, e.g.

```
jobDBInspector := jobsdb.HandleInspector{&jobDB}
require.Equal(t, (2, jobDBInspector.DSListSize())
```

Reply:
I wasn't comfortable too exposing that function. But din't have an option. The above wrapping is neat. Will make the change.

Reply:
done